accounts-db/src/accounts_db.rs

```diff
@@ -2742,86 +2755,95 @@ impl AccountsDb {
     /// 1. one of the pubkeys in the store has account info to a store whose store count is not going to zero
     /// 2. a pubkey we were planning to remove is not removing all stores that contain the account
     fn calc_delete_dependencies(
-        purges: &HashMap<Pubkey, (SlotList<AccountInfo>, RefCount)>,
+        purges: &Vec<RwLock<HashMap<Pubkey, CleaningInfo>>>,
         store_counts: &mut HashMap<Slot, (usize, HashSet<Pubkey>)>,
         min_slot: Option<Slot>,
     ) {
         // Another pass to check if there are some filtered accounts which
         // do not match the criteria of deleting all appendvecs which contain them
         // then increment their storage count.
         let mut already_counted = IntSet::default();
-        for (pubkey, (account_infos, ref_count_from_storage)) in purges.iter() {
```

Comment:
i suggest a separate pr that renames account_infos -> slot_list, ref_count_from_storage -> ref_count.
This will eliminate some noise in this pr and let this pr focus on the much more interesting issues at hand. Get that pr merged, then rebase this on master.